selfservice/strategy/oidc/provider_linkedin.go

```diff
@@ -0,0 +1,210 @@
+	resp, err := client.Do(req)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	defer resp.Body.Close()
+	if err := json.NewDecoder(resp.Body).Decode(result); err != nil {
+		return errors.WithStack(err)
+	}
+
+	return nil
+}
+
+func (l *ProviderLinkedIn) Profile(client *retryablehttp.Client) (*Profile, error) {
+	var profile Profile
+
+	if err := l.ApiGetCall(client, ProfileUrl, &profile); err != nil {
+		return nil, errors.WithStack(err)
+	}
+	return &profile, nil
+}
+
+func (l *ProviderLinkedIn) Email(client *retryablehttp.Client) (*EmailAddress, error) {
+	var emailaddress EmailAddress
+
+	if err := l.ApiGetCall(client, EmailUrl, &emailaddress); err != nil {
+		return nil, errors.WithStack(err)
+	}
+
+	return &emailaddress, nil
+}
+
+func (l *ProviderLinkedIn) Introspection(client *retryablehttp.Client, result interface{}, exchange *oauth2.Token) error {
+	form := url.Values{"client_id": {l.config.ClientID}, "client_secret": {l.config.ClientSecret}, "token": {exchange.AccessToken}}
+	req, err := retryablehttp.NewRequest(http.MethodPost, string(IntrospectionURL), strings.NewReader(form.Encode()))
+	if err != nil {
+		return errors.WithStack(err)
+	}
+	req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	req.PostForm = form
+	resp, err := client.Do(req)
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	defer resp.Body.Close()
+	if err := json.NewDecoder(resp.Body).Decode(result); err != nil {
+		return errors.WithStack(err)
+	}
+
+	return nil
+}
+
+func (l *ProviderLinkedIn) VerifyScopes(client *retryablehttp.Client, exchange *oauth2.Token) error {
+	var introspection Introspection
+
+	if err := l.Introspection(client, &introspection, exchange); err != nil {
+		return errors.WithStack(err)
+	}
+	grantedScopes := stringsx.Splitx(introspection.Scope, ",")
+	for _, check := range l.Config().Scope {
+		if !stringslice.Has(grantedScopes, check) {
+			return errors.WithStack(ErrScopeMissing)
+		}
+	}
+
+	return nil
+}
+
+func (l *ProviderLinkedIn) Claims(ctx context.Context, exchange *oauth2.Token, query url.Values) (*Claims, error) {
+
+	var profile *Profile
+	var emailaddress *EmailAddress
+
+	o, err := l.OAuth2(ctx)
+	if err != nil {
+		return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("%s", err))
+	}
+
+	client := l.reg.HTTPClient(ctx, httpx.ResilientClientWithClient(o.Client(ctx, exchange)))
+	if err = l.VerifyScopes(client, exchange); err != nil {
+		return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("%s", err))
+	}
+	profile, err = l.Profile(client)
+	if err != nil {
+		return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("%s", err))
+	}
+	emailaddress, err = l.Email(client)
+	if err != nil {
+		return nil, errors.WithStack(herodot.ErrInternalServerError.WithReasonf("%s", err))
+	}
+
+	claims := &Claims{
+		Email:     emailaddress.Elements[0].Handle.EmailAddress,
+		Name:      fmt.Sprintf("%s %s", profile.LocalizedFirstName, profile.LocalizedLastName),
```

Comment:
We only want to include fields the upstream provides us - data changes we can do in the jsonnet itself:

```suggestion
```